app/models/mixins/provider_proxy_mixin.rb

```diff
@@ -0,0 +1,13 @@
+module ProviderProxyMixin
+  extend ActiveSupport::Concern
+
+  def http_proxy_uri
+    proxy = VMDB::Util.http_proxy_uri("#{emstype}_http_proxy".to_sym)
```

Comment:
yeah, sorry, my late night brain dump reads a bit unclear :smile: 

In settings.yml we have currently 2 ways for provider specific settings.

* suffix: like `event_catcher_vmware` 
* nested: like `ems_refresh` -> `ec2`

This would introduce a third, prefix: `vmware_http_proxy`

I would go with the nested approach and move that logic to `VMDB::Util.http_proxy_uri` 

```ruby
VMDB::Util.http_proxy_uri(ems_type)

def self.http_proxy_uri(proxy_config = nil)
  proxy = Settings.http_proxy[subsystem] || Settings.http_proxy
```

Then we can have in settings.yml
```yaml
:http_proxy:
  :host:
  :password:
  :port:
  :user:
  :ec2:
     :host:
     :password:
     :port:
     :user:
```

It would be even nicer with defaults like [here](https://github.com/ManageIQ/manageiq/blob/faec76397e8e77d65e98432d8d46e38357632832/config/settings.yml#L1101) - but I guess that would break backward compatability